go/arrow/avro/schema.go

```diff
@@ -0,0 +1,409 @@
+			return arrow.Field{Name: node.name, Type: arrow.ListOf(avroLogicalToArrowField(n).Type)}
+		}
+	}
+	// Avro "enum" field type = Arrow dictionary type
+	if i, ok := node.ofType.(map[string]interface{})["symbols"]; ok {
+		symbols := make(map[string]string)
+		for index, symbol := range i.([]interface{}) {
+			k := strconv.FormatInt(int64(index), 10)
+			symbols[k] = symbol.(string)
+		}
+		var dt arrow.DictionaryType = arrow.DictionaryType{IndexType: arrow.PrimitiveTypes.Uint64, ValueType: arrow.BinaryTypes.String, Ordered: false}
+		sl := len(symbols)
+		switch {
+		case sl <= math.MaxUint8:
+			dt.IndexType = arrow.PrimitiveTypes.Uint8
+		case sl > math.MaxUint8 && sl <= math.MaxUint16:
+			dt.IndexType = arrow.PrimitiveTypes.Uint16
+		case sl > math.MaxUint16 && sl <= math.MaxUint32:
+			dt.IndexType = arrow.PrimitiveTypes.Uint32
+		}
+		return arrow.Field{Name: node.name, Type: &dt, Nullable: true, Metadata: arrow.MetadataFrom(symbols)}
+	}
+	// Avro "fixed" field type = Arrow FixedSize Primitive BinaryType
+	if i, ok := node.ofType.(map[string]interface{})["size"]; ok {
+		return arrow.Field{Name: node.name, Type: &arrow.FixedSizeBinaryType{ByteWidth: int(i.(float64))}}
+	}
+	// Avro "map" field type
+	if i, ok := node.ofType.(map[string]interface{})["values"]; ok {
+		return arrow.Field{Name: node.name, Type: arrow.MapOf(arrow.BinaryTypes.String, AvroPrimitiveToArrowType(i.(string)))}
+	}
+	// Avro "record" field type
+	if _, f := node.ofType.(map[string]interface{})["fields"]; f {
+		for _, field := range node.ofType.(map[string]interface{})["fields"].([]interface{}) {
+			n.fields = append(n.fields, field.(map[string]interface{}))
+		}
+		s := iterateFields(n.fields)
+		return arrow.Field{Name: n.name, Type: arrow.StructOf(s...)}
+	}
+	return arrow.Field{}
+}
+
+// AvroPrimitiveToArrowType returns the Arrow DataType equivalent to a
+// Avro primitive type.
+//
+// NOTE: Arrow Binary type is used as a catchall to avoid potential data loss.
+func AvroPrimitiveToArrowType(avroFieldType string) arrow.DataType {
+	switch avroFieldType {
+	// int: 32-bit signed integer
+	case "int":
+		return arrow.PrimitiveTypes.Int32
+	// long: 64-bit signed integer
+	case "long":
+		return arrow.PrimitiveTypes.Int64
+	// float: single precision (32-bit) IEEE 754 floating-point number
+	case "float":
+		return arrow.PrimitiveTypes.Float32
+	// double: double precision (64-bit) IEEE 754 floating-point number
+	case "double":
+		return arrow.PrimitiveTypes.Float64
+	// bytes: sequence of 8-bit unsigned bytes
+	case "bytes":
+		return arrow.BinaryTypes.Binary
+	// boolean: a binary value
+	case "boolean":
+		return arrow.FixedWidthTypes.Boolean
+	// string: unicode character sequence
+	case "string":
+		return arrow.BinaryTypes.String
+	// fallback to binary type for any unsupported type
+	default:
+		return arrow.BinaryTypes.Binary
```

Comment:
I'd prefer an error with `arrow.ErrNotImplemented` for any unsupported type

Reply:
The only unimplemented type is Duration logical type which is unimplemented in linkedin/goavro but is implemented in hamba/avro. If we change to hamba for the reader then all types would be covered and we can just get rid of the fallback.

Reply:
hamba/avro supports duration logical type, all avro primitive and logical types are now implemented.